### PR TITLE
Disable social buttons for users under 13

### PR DESCRIPTION
**Pull request by maddiedierker**
Depends on #26154.

Fixes [this issue in dance party](https://github.com/code-dot-org/dance-party/issues/287).

Since we ask signed out users for their age before using Dance Party, we can use that age to decide whether to hide social share (Facebook and Twitter) buttons. Now, we will hide social share for all users under 13 (whether or not they are authenticated).

### Finish dialog
**Before:**
screen shot 2018-11-20 at 10 21 26 am

**After:**
screen shot 2018-11-20 at 10 19 27 am

### Share dialog
**Before:**
screen shot 2018-11-20 at 10 21 33 am

**After:**
screen shot 2018-11-20 at 10 19 38 am

### CODE REVIEW

File: dashboard/test/ui/features/dance/social_share.feature [hunk 1]
```diff
@@ -0,0 +1,62 @@
+# Testing that the social share buttons are shown/hidden in the correct scenarios
+# in Dance Party.
+
+Feature: Social share buttons
+  Scenario: Signed in 13+ user sees social share buttons
+    Given I create a teacher named "Hermoine"
+    And I am on "http://studio.code.org/s/dance/stage/1/puzzle/13"
```
Comment (maddiedierker) [inline dashboard/test/ui/features/dance/social_share.feature]: what's the best way to check that this line will be okay in all environments? i recall trying to use a script that isn't available in either staging or test in the past, and that causing the test to break in that environment

Comment (islemaster) [reply dashboard/test/ui/features/dance/social_share.feature]: If it's in this list it'll be available for UI tests in all environments:

https://github.com/code-dot-org/code-dot-org/blob/2935aba881ca86908188913acf05e682a778e0c1/dashboard/lib/tasks/seed.rake#L21

File: dashboard/test/ui/features/dance/social_share.feature [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+  Scenario: Signed in user under 13 does not see social share buttons
+    Given I create a young student named "Harry"
+    And I am on "http://studio.code.org/s/dance/stage/1/puzzle/13"
+    And I rotate to landscape
+    And I wait for the page to fully load
+
+    And I wait until I see selector ".project_share"
+    When I click ".project_share"
+    And I wait for 10 seconds
```
Comment (islemaster) [inline dashboard/test/ui/features/dance/social_share.feature]: Should this be changed to 3 seconds like the other two scenarios?

I'd normally suggest we wait for something other than an set time, but in this case with the reachability check that runs I'm not sure what we'd wait for.

Comment (maddiedierker) [reply dashboard/test/ui/features/dance/social_share.feature]: 😱 i was testing something and never changed it back to 3 seconds!

**Comment by Erin007** [comment]
Based on screenshots: why are we hiding the "send to phone" button in the finish dialog, but not in the share dialog?

**Comment by maddiedierker** [comment]
@Erin007 that's a good question! the existing functionality for signed in users works the same way, but i honestly don't know why. 

@ryansloan -- signed in users under 13 currently see the "share to phone" button in the share dialog, but it's hidden in the finish dialog (screenshots below). should this button be shown or hidden for users under 13?

**Share dialog (user under 13 sees "share to phone"):**
screen shot 2018-11-15 at 10 09 10 pm

**Finish dialog (user under 13 does not see "share to phone"):**
screen shot 2018-11-15 at 10 09 21 pm

**Comment by ryansloan** [comment]
@madelynkasula Send to phone should be visible for under 13, as confirmed by Privacy Czar. If we can show it in Finish as part of this that would be great.

**Comment by maddiedierker** [comment]
@ryansloan will do! @Erin007, thanks for bringing that up

**Pull request merged**